[Update] supported third lib (#10260)
pingcap/docs: develop/dev-guide-third-party-support.md

---
title: Third-Party Tools Support by TiDB
summary: Learn about third-party tools support by TiDB.
---

# Third-Party Tools Support by TiDB

> **Note:**
>

> > Only some of the supported third-party tools are listed here. However, PingCAP has no way of knowing whether all third-party tools use unsupported features and thus cannot guarantee compatibility.

Reviewer: ```suggestion
> This document only lists some common third-party tools supported by TiDB. Third-party tools that are not listed do not mean that they are not supported. PingCAP has no way of knowing whether they use features that TiDB does not support and thus compatibility cannot be guaranteed.
```

> > This document only lists some common third-party tools supported by TiDB. Third-party tools that are not listed do not mean that they are not supported. PingCAP has no way of knowing whether they use features that TiDB does not support and thus compatibility cannot be guaranteed.

Reviewer 2: ```suggestion
> This document only lists common third-party tools supported by TiDB. Some other third-party tools are not listed, not because they are not supported, but because PingCAP is not sure whether they use features that are incompatible with TiDB.
```

| Java                  | [MyBatis](https://mybatis.org/mybatis-3/)                                                                                                                                     | v3.5.10     | Full | N/A                                                    | [Build a Simple CRUD App with TiDB and Java](/develop/dev-guide-sample-application-java.md)                                                                            |
| Java                  | [Spring Data JPA](https://spring.io/projects/spring-data-jpa/) | 2.7.2 | Full | N/A                                                    |  [Build a TiDB Application Using Spring Boot](/develop/dev-guide-sample-application-spring-boot.md)   |
| Java                  | [jOOQ](https://github.com/jOOQ/jOOQ)                                                                                                                                     | v3.16.7 (Open Source)     | Full | N/A                                                    | N/A                                                                            |

> | Ruby | [ActiveRecord](https://guides.rubyonrails.org/active_record_basics.html) | v7.0 | Full | N/A | N/A |

Reviewer: ```suggestion
| Ruby | [Active Record](https://guides.rubyonrails.org/active_record_basics.html) | v7.0 | Full | N/A | N/A |
```
